src/Foundatio.Storage.SshNet/Storage/SshNetFileStorage.cs

```diff
@@ -13,384 +14,432 @@
+        foreach (var file in files.Where(f => f.IsRegularFile || f.IsDirectory).OrderBy(f => f.IsRegularFile).ThenBy(f => f.Name)) {
+            if (cancellationToken.IsCancellationRequested) {
+                _logger.LogDebug("Returning partial file list: Cancellation requested");
                 return;
             }
 
-            foreach (var file in files.Where(f => f.IsRegularFile || f.IsDirectory).OrderBy(f => f.IsRegularFile).ThenBy(f => f.Name)) {
-                if (cancellationToken.IsCancellationRequested)
-                    return;
+            if (recordsToReturn.HasValue && list.Count >= recordsToReturn)
+                break;
 
-                if (recordsToReturn.HasValue && list.Count >= recordsToReturn)
-                    break;
-
-                if (file.IsDirectory) {
-                    if (file.Name is "." or "..")
-                        continue;
-
-                    await GetFileListRecursivelyAsync(String.Concat(prefix, "/", file.Name), pattern, list, recordsToReturn, cancellationToken).AnyContext();
-                    continue;
-                }
-
-                if (!file.IsRegularFile)
+            if (file.IsDirectory) {
+                if (file.Name is "." or "..")
                     continue;
 
-                string path = file.FullName.TrimStart('/');
-                if (pattern != null && !pattern.IsMatch(path))
-                    continue;
-
-                list.Add(new FileSpec {
-                    Path = path,
-                    Created = file.LastWriteTimeUtc,
-                    Modified = file.LastWriteTimeUtc,
-                    Size = file.Length
-                });
+                await GetFileListRecursivelyAsync(String.Concat(prefix, "/", file.Name), pattern, list, recordsToReturn, cancellationToken).AnyContext();
+                continue;
             }
-        }
-
-        private ConnectionInfo CreateConnectionInfo(SshNetFileStorageOptions options) {
-            if (String.IsNullOrEmpty(options.ConnectionString))
-                throw new ArgumentNullException(nameof(options.ConnectionString));
 
-            if (!Uri.TryCreate(options.ConnectionString, UriKind.Absolute, out var uri) || String.IsNullOrEmpty(uri?.UserInfo))
-                throw new ArgumentException("Unable to parse connection string uri", nameof(options.ConnectionString));
+            if (!file.IsRegularFile)
+                continue;
 
-            string[] userParts = uri.UserInfo.Split(new [] { ':' }, StringSplitOptions.RemoveEmptyEntries);
-            string username = Uri.UnescapeDataString(userParts.First());
-            string password = Uri.UnescapeDataString(userParts.Length > 1 ? userParts[1] : String.Empty);
-            int port = uri.Port > 0 ? uri.Port : 22;
+            string path = file.FullName.TrimStart('/');
+            if (pattern != null && !pattern.IsMatch(path)) {
+                _logger.LogTrace("Skipping {Path}: Doesn't match pattern", path);
```

Comment:
This will be a massive firehose in a system with a large amount of files. This could potentially spit out 1000s and 1000s of log messages. I guess this is trace level so you kind of expect it to be a firehose, but do you really think this is adding value?

Reply:
Yeah, you are doing a regex pattern matching. I kinda want to know if something is skipped.